app/controllers/application_controller.rb

```diff
@@ -1,2 +1,9 @@
 class ApplicationController < ActionController::Base
+  def after_sign_in_path_for(resource)
+    if current_client && (current_client.client_profile.nil? || current_client.client_profile.invalid?)
```

Comment:
Aqui acho que você poderia usar o `safe navigator` pra simplificar esse código, não tenho certeza que vai funcionar mas acho que daria pra fazer algo como:

```suggestion
    if current_client&.client_profile&.invalid?
```

Porque se não tiver profile nem chamaria o invalid? então seria o mesmo que retornar nulo

Reply:
Então Henrique, aproveitando o gancho do que você comentou mais abaixo pois achei mais legível do que dessa forma, fiz assim:

def after_sign_in_path_for(resource)
    if current_client&.client_profile?
      stored_location_for(resource) || root_path
    else
      stored_location_for(resource) || new_client_profile_path
    end
  end
  
  E daí no model client eu adicionei o método client_profile? como você sugeriu!

Reply:
👌